src/main/groovy/com/vanniktech/android/javadoc/Generation.groovy

```diff
@@ -158,12 +175,15 @@ class Generation implements Plugin<Project> {
 
     private void addDeleteJavadocTaskToCleanTaskIn(final Project project, variant) {
         String taskName = genDeleteTaskName(project, variant)
-        if (project.tasks.findByPath(taskName)) {
+        if (findTask(project, taskName)) {
```

Comment:
This is inconsistent with `createRootTask`. How come? I'd expect to not use `findTask` here or also use `findTask` in `createRootTask`.

Reply:
findTask is just a helper that uses new TaskAvoidance API (`project.tasks.named()`)

I am using this to avoid creating a task that already exists for some reason i don't know